tests/golos.publication_tests.cpp

```diff
@@ -0,0 +1,439 @@
+                          ("posting_battery", mvo()("charge", 2500))
+                          ("battery_of_votes", mvo()("charge", 0))
+                          ("limit_battery_posting", mvo()("charge", 10000))
+                          ("limit_battery_comment", mvo()("charge", 0))
+                          ("limit_battery_of_votes", mvo()("charge", 0))
+                          );
+    produce_blocks(10);
+
+    BOOST_REQUIRE_EQUAL(error("assertion failure with message: Battery overrun"), golos_publication_tester::create_post(
+                            N(golos.pub),
+                            "permlink1",
+                            N(),
+                            "",
+                            333,
+                            "payouttype",
+                            {{N(beneficiary), 777}},
+                            "paytype",
+                            "headerpost",
+                            "bodypost",
+                            "languagepost",
+                            {{"tag"}},
+                            "jsonmetadata"));
+
+    account_battery = get_battery(N(golos.pub));
+    require_equal_battery(account_battery, mvo()
+                          ("posting_battery", mvo()("charge", 2500))
+                          ("battery_of_votes", mvo()("charge", 0))
+                          ("limit_battery_posting", mvo()("charge", 10000))
+                          ("limit_battery_comment", mvo()("charge", 0))
+                          ("limit_battery_of_votes", mvo()("charge", 0))
+                          );
+    produce_blocks(21);
+
+    BOOST_REQUIRE_EQUAL(success(), golos_publication_tester::create_post(
+                            N(golos.pub),
+                            "permlink2",
+                            N(),
+                            "",
+                            333,
+                            "payouttype",
+                            {{N(beneficiary), 777}},
+                            "paytype",
+                            "headerpost",
+                            "bodypost",
+                            "languagepost",
+                            {{"tag"}},
+                            "jsonmetadata"));
+    produce_blocks(21);
+
+    BOOST_REQUIRE_EQUAL(success(), golos_publication_tester::create_post(
+                            N(golos.pub),
+                            "permlink3",
+                            N(),
+                            "",
+                            333,
+                            "payouttype",
+                            {{N(beneficiary), 777}},
+                            "paytype",
+                            "headerpost",
+                            "bodypost",
+                            "languagepost",
+                            {{"tag"}},
+                            "jsonmetadata"));
+    produce_blocks(21);
+
+    BOOST_REQUIRE_EQUAL(success(), golos_publication_tester::create_post( // TODO no check in smart-contract
```

Comment:
it's not clear, what this sequential actions test. maybe add battery check at this point? or check that next `create_post` fails (if it must)